woocommerce/woocommerce-android
# [Mobile Payments] Track country and currency on payments flow collect event

Closes: #10527


### Description

The PR:
* Moves `CardReaderTracking` to the upper payments package
* Renames it to `PaymentsFlowTracker` to reflect it's new purpose
* Moves tracking of the following events in there so it will add properties that we use for other payments-related events:
  * `PAYMENTS_FLOW_COLLECT`
  * `PAYMENTS_FLOW_COMPLETED`
  * `PAYMENTS_FLOW_CANCELED`
  * `PAYMENTS_FLOW_FAILED`

### Testing instructions

* Try both payment flows:
  * Order lists -> create order -> collect payment 
  * More -> payments -> collect payment
* Try starting (collect), completing, canceling or failing the flow
* Using logcat with a "track" filter, notice the events with `currency` and `country` properties in it  (The `canceled` event tracked only when the "back" button is clicked on the payment method selection screen during payment for the order)

```
Tracked: payments_flow_collect, Properties: {"source":"payment_method","order_id":2954,"flow":"simple_payment","payment_method":"card","card_reader_type":"external","plugin_slug":"woocommerce-payments","country":"US","currency":"USD","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","store_id":"3ee2abc7-2ac9-4e18-85ae-714aeb76377a","is_debug":true,"site_url":"https:\/\/site-for-woocommerce12a3fasdf45dfs6789.mystagingwebsite.com"}

Tracked: payments_flow_failed, Properties: {"source":"payment_method","flow":"simple_payment","plugin_slug":"woocommerce-payments","country":"US","currency":"USD","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","store_id":"3ee2abc7-2ac9-4e18-85ae-714aeb76377a","is_debug":true,"site_url":"https:\/\/site-for-woocommerce12a3fasdf45dfs6789.mystagingwebsite.com"}

Tracked: payments_flow_completed, Properties: {"flow":"simple_payment","payment_method":"cash","order_id":2954,"amount":"$1.15","amount_normalized":115,"plugin_slug":"woocommerce-payments","country":"US","currency":"USD","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","store_id":"3ee2abc7-2ac9-4e18-85ae-714aeb76377a","is_debug":true,"site_url":"https:\/\/site-for-woocommerce12a3fasdf45dfs6789.mystagingwebsite.com"}

Tracked: payments_flow_canceled, Properties: {"source":"payment_method","flow":"order_payment","plugin_slug":"woocommerce-payments","country":"US","currency":"USD","blog_id":154993204,"is_wpcom_store":false,"was_ecommerce_trial":false,"plan_product_slug":"jetpack_premium","store_id":"3ee2abc7-2ac9-4e18-85ae-714aeb76377a","is_debug":true,"site_url":"https:\/\/site-for-woocommerce12a3fasdf45dfs6789.mystagingwebsite.com"}
```
